### PR TITLE
Update system proxy management to run FFI calls on isolate

### PR DESCRIPTION
**Pull request by atavism**
This PR updates the desktop FFI code for managing the system proxy to run on a separate isolate, improving overall stability. It resolves some issues we've been seeing with conflicting signal handling between Dart and Go by moving the FFI calls to a separate thread, which helps to avoid problematic interactions with the Go runtime signal stack

Working well for me on macOS

### CODE REVIEW

**Comment by jigar-f** [comment]
@atavism I made few more changes, Do take look

**Comment by jigar-f** [comment]
Also, there is one more issue, somehow bandwidth is not getting all data from desktop.

File: lib/core/service/lantern_ffi_service.dart [hunk 1]
```diff
@@ -61,6 +68,57 @@ class LanternFFI {
...
+    if (!_isolateInitialized.isCompleted) {
+      await _initializeSystemProxyIsolate();
+    }
+
+    final responsePort = ReceivePort(); // Port to receive isolate's response
+    _proxySendPort?.send([vpnStatus, responsePort.sendPort]);
+
+    // Listen for the result (success or error)
+    final message = await responsePort.first;
+    if (message == "done") {
```
Comment (jigar-f) [inline lib/core/service/lantern_ffi_service.dart:91]: Before we are not checking error and done from isolate now we are doing it.

File: lib/features/vpn/vpn_switch.dart [hunk 2]
```diff
@@ -49,42 +51,68 @@ class _VPNSwitchState extends State<VPNSwitch> {
         });
       });
     } else {
-      // This ui for desktop
-      return AdvancedSwitch(
```
Comment (jigar-f) [inline lib/features/vpn/vpn_switch.dart:53]: I had to remove AdvancedSwitch and use new one, reason being there was issue with control.

File: lib/features/vpn/vpn_switch.dart [hunk 3]
```diff
@@ -110,4 +138,16 @@ class _VPNSwitchState extends State<VPNSwitch> {
       );
     }
   }
+
+  Future<void> vpnProcessForDesktop(String vpnStatus) async {
+    final vpnNotifier = Provider.of<VPNChangeNotifier>(context, listen: false);
```
Comment (jigar-f) [inline lib/features/vpn/vpn_switch.dart:143]: Now we are checking for errors, if errors are then we show the message and set the status.

File: desktop/lib.go [hunk 4]
```diff
@@ -135,8 +135,13 @@ func saveUserSalt(salt []byte) {
 }
 
 //export sysProxyOn
-func sysProxyOn() {
-	go a.SysproxyOn()
+func sysProxyOn() *C.char {
+	err := a.SysproxyOn()
```
Comment (jigar-f) [inline desktop/lib.go:139]: Now we are returning error on flutter end and showing error to user

File: lib/features/vpn/vpn_switch.dart (hunk 2 above)
Comment (atavism) [reply lib/features/vpn/vpn_switch.dart:53]: What issue are you experiencing with the control..? It works fine as is for me:

https://github.com/user-attachments/assets/88cafe37-2473-47f3-849f-bb4d84b71dbf

File: lib/core/service/lantern_ffi_service.dart (hunk 1 above)
Comment (atavism) [reply lib/core/service/lantern_ffi_service.dart:91]: Just to give some context: I thought we could avoid some redundant response checking by sending the status update (if setting the system proxy succeeded or failed) via the websocket channel afterwards (actually, I don't think we need to do that anymore and can simplify things further with this setup).

Using a ReceivePort here to check if the isolate _received/processed_ the request though definitely makes sense. Thanks for making the change!

File: lib/features/vpn/vpn_switch.dart (hunk 2 above)
Comment (atavism) [reply lib/features/vpn/vpn_switch.dart:53]: I definitely like the change to use animated_toggle_switch now so I think the control issues are irrelevant.

**Comment by atavism** [comment]
> Also, there is one more issue, somehow bandwidth is not getting all data from desktop.

Fixed.

**Pull request merged**